main.cpp

```diff
@@ -43,17 +43,29 @@ int main(int argc, char **argv)
             }
             case 'I': { // include path
                 const char * const value = arg[2] ? (argv[i] + 2) : argv[++i];
-                dui.includePaths.push_back(value);
+                dui.addIncludePath(value, /* legacy= */ false);
```

Comment:
if "legacy" is used simplecpp will not just skip over the new handling it will copy all the include paths (and as far as I see this is repeated). This solution will introduce a performance penalty.

We have had major performance problems and we managed to improve the situation but this I fear adds cpu overhead.

I envision that in Cppcheck many users will only use `-I` flags and I would like to be able to get optimal performance by choosing "legacy=true"